api/team.go

```diff
@@ -358,6 +360,11 @@ func LeaveTeam(team *model.Team, user *model.User) *model.AppError {
 		return uua.Err
 	}
 
+	// delete the preferences that set the last channel used in the team and other team specific preferences
+	if result := <-Srv.Store.Preference().DeleteCategory(user.Id, team.Id); result.Err != nil {
```

Comment:
For anyone else reading this, `LeaveTeam` is called when a user is removed from the team